src/icons.rs

```diff
@@ -20,16 +20,19 @@ impl From<&'static str> for Icon {
     }
 }
 
+impl From<Vec<&'static str>> for Icon {
```

Comment:
Since the memory from `Vec` is not reused, this can be `&[&str]` instead of `Vec<&'static str>`.

Reply:
When I tried that I got:

```
error[E0277]: the trait bound `Icon: From<[&str; 6]>` is not satisfied
   --> src/util.rs:159:41
    |
159 |           $map.insert($key.into(), $value.into());
    |                                           ^^^^ the trait `From<[&str; 6]>` is not implemented for `Icon`
    |
   ::: src/icons.rs:32:14
    |
32  |           Self(map! {
    |  ______________-
33  | |             "backlight" => "BRIGHT",
34  | |             "bat" => "BAT",
35  | |             "bat_charging" => "CHG",
...   |
106 | |             "xrandr" => "SCREEN"
107 | |         })
    | |_________- in this macro invocation
    |
    = help: the following other types implement trait `From<T>`:
              <Icon as From<&'static str>>
              <Icon as From<&[&str]>>
    = note: required for `[&str; 6]` to implement `std::convert::Into<Icon>`
    = note: this error originates in the macro `map` (in Nightly builds, run with -Z macro-backtrace for more info)
    ```

Reply:
I tried `["..", ".."]` and `&["..", ".."]`, and neither worked, but `["..", ".."].as_slice()` and `["..", ".."].as_ref()` work.

I thought `&["..", ".."]` and  `["..", ".."].as_ref()` were equivalent.

Reply:
Oh, we could do this:

```rust
impl<const N: usize> From<[&str; N]> for Icon {
    fn from(value: [&str; N]) -> Self {
        Self::Progression(value.iter().map(|s| s.to_string()).collect())
    }
}
```
or
```rust
impl<const N: usize> From<&[&str; N]> for Icon {
    fn from(value: &[&str; N]) -> Self {
        Self::Progression(value.iter().map(|s| s.to_string()).collect())
    }
}
```

Using `["..", ".."]` and `&["..", ".."]` respectively. The former looks cleaner, but if it's better (performance?) to pass the ref we can use the latter.

Reply:
> but if it's better (performance?) to pass the ref

I think in our case of 6 elements it doesn't matter :)